controller/contactsController.js

```diff
@@ -0,0 +1,128 @@
+const service = require("../service/index");
+const {
+  validateAddContact,
+  validateUpdateContact,
+  validateUpdateFavorite,
+} = require("../tools/validator");
+
+const get = async (req, res, next) => {
+  try {
+    const contactsList = await service.getAllContacts();
+    console.log(contactsList);
+    res.json({ status: 200, body: contactsList });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const getById = async (req, res, next) => {
+  const { contactId } = req.params;
+  try {
+    const getContact = await service.getContactById(contactId);
+    console.log(getContact);
+    if (getContact) {
+      return res.json({ status: 200, data: getContact });
+    } else {
+      return res.json({ status: 404, message: `Not found contact id: ${contactId}` });
+    }
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const createContact = async (req, res, next) => {
+  const { name, email, phone, favorite = false } = req.body;
+  const { error } = validateAddContact(req.body);
+
+  try {
+    if (error) {
+      console.log(error);
+      return res.json({ status: 400, message: "Missing fields" });
+    }
+    const newContact = await service.createContact({
+      name,
+      email,
+      phone,
+      favorite,
+    });
+    res.json({ status: 201, message: "Add new contact", data: newContact });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const removeContact = async (req, res, next) => {
+  const { contactId } = req.params;
+
+  try {
+    const contactToRemove = await service.removeContact(contactId);
+    if (contactToRemove) {
+      res.json({
+        status: 200,
+        message: "Contact deleted",
+        data: contactToRemove,
+      });
+    } else {
+      res.json({ status: 404, message: `Not found contact id: ${contactId}` });
+    }
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const updateContact = async (req, res, next) => {
+  const { contactId } = req.params;
+  const body = req.body;
+  const { error } = validateUpdateContact(req.body);
+
+  try {
+    if (error) {
+      console.log(error);
+      return res.json({ status: 400, message: "Missing fields" });
+    }
+    const renameContact = await service.updateContact(contactId, body);
+    if (renameContact) {
+      res.json({ status: 200, message: "Contact update", data: renameContact });
+    } else {
+      res.json({ status: 404, message: `Not found contact id: ${contactId}` });
+    }
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const updateFavorite = async (req, res, next) => {
+  const { contactId } = req.params;
+  const { favorite = false } = req.body;
+  const { error } = validateUpdateFavorite(req.body);
+
+  try {
+    if (error) {
+      console.log(error);
+      return res.json({ status: 400, message: "Missing field favorite" });
```

Comment:
Jesli chcesz byc troche bardziej "efficient", to `message` mozesz skrocic do `msg`. Bedziesz przesylal znaczaco mniej danych z serwera, jesli policzysz ilosc routes razy ilosc avg requestow razy ilosc userow (jesli userzy ida w tysiace/miliony, to zaczyna miec spore znaczenie na kilku plaszczyznach: czas wysylki, obciazenie serwera, srodowisko (zuzycie energii)).

Reply:
poprawione